pyomo/dae/plugins/colloc.py

```diff
@@ -470,6 +459,23 @@ def _transformBlock(self, block, currentds):
                 add_discretization_equations(d.parent_block(), d)
                 d.parent_block().reclassify_component_type(d, Var)
 
+                # Keep track of any reclassified DerivativeVar components so
+                # that the Simulator can easily identify them if the model
+                # is simulated after discretization
+                # TODO: Update the discretization transformations to use
+                # a Block to add things to the model and store discretization
+                # information. Using a list for now because the simulator
+                # does not yet support models containing active Blocks
+                reclassified_list = getattr(block,
+                                            '_pyomo_dae_reclassified_derivativevars',
+                                            None)
+                if reclassified_list is None:
+                    block._pyomo_dae_reclassified_derivativevars = list()
+                    reclassified_list = \
+                        block._pyomo_dae_reclassified_derivativevars
+
```

Comment:
@jsiirola see the TODO comment directly above this section of code. I intend to put all of this transformation information in a block but this requires significant changes in the Simulator. Adding a list directly to the block being discretized is a quick (and temporary) fix for #531. Because this is a temporary solution which will be replaced in part 2 of this overhaul, I would prefer not to add the attribute check you're asking for here. I am planning on including it in the next PR along with using the `unique_component_name` function.

Reply:
Sounds good to me.